src/cli/services/__tests__/linter.test.ts

```diff
@@ -0,0 +1,386 @@
+          },
+          source: expect.stringContaining('/__tests__/__fixtures__/gh-474/common.yaml'),
+        }),
+      ]),
+    );
+  });
+
+  it('should handle relative path to a document #2', async () => {
+    const results = await run('lint src/__tests__/__fixtures__/gh-474/spec-2.yaml');
+
+    expect(results).toEqual(
+      expect.not.arrayContaining([
+        expect.objectContaining({
+          code: 'invalid-ref',
+        }),
+      ]),
+    );
+
+    expect(results).toEqual(
+      expect.arrayContaining([
+        expect.objectContaining({
+          code: 'oas3-schema',
+          range: {
+            end: {
+              character: 14,
+              line: 5,
+            },
+            start: {
+              character: 10,
+              line: 5,
+            },
+          },
+          source: expect.stringContaining('/__tests__/__fixtures__/gh-474/common.yaml'),
+        }),
+      ]),
+    );
+  });
+
+  describe('when document is local file', () => {
+    describe('and the file is expected to have no warnings', () => {
+      const document = resolve(__dirname, '__fixtures__/openapi-3.0-valid.yaml');
+
+      it('outputs no issues', () => {
+        return expect(run(`lint ${document}`)).resolves.toEqual([]);
+      });
+    });
+
+    describe('and the file is expected to trigger oas3 warnings', () => {
+      const document = resolve(__dirname, '__fixtures__/openapi-3.0-no-contact.yaml');
+
+      it('outputs warnings', async () => {
+        const output = await run(`lint ${document}`);
+        expect(logSpy).toHaveBeenCalledWith('OpenAPI 3.x detected');
+        expect(output).toEqual(
+          expect.arrayContaining([
+            expect.objectContaining({ code: 'api-servers' }),
+            expect.objectContaining({ code: 'info-contact' }),
+          ]),
+        );
+      });
+
+      describe('and --skip-rule=info-contact is set', () => {
+        it('output other warnings but not info-contact', async () => {
+          const output = await run(`lint --skip-rule=info-contact ${document}`);
+
+          expect(output).toEqual(expect.arrayContaining([expect.objectContaining({ code: 'api-servers' })]));
+          expect(output).toEqual(expect.not.arrayContaining([expect.objectContaining({ code: 'info-contact' })]));
+        });
+      });
+
+      describe('and --skip-rule=info-contact --skip-rule=api-servers is set', () => {
+        it('outputs neither info-contact or api-servers', async () => {
+          const output = await run(`lint --skip-rule=info-contact --skip-rule=api-servers ${document}`);
+
+          expect(output).toEqual(expect.not.arrayContaining([expect.objectContaining({ code: 'info-contact' })]));
+          expect(output).toEqual(expect.not.arrayContaining([expect.objectContaining({ code: 'api-servers' })]));
+        });
+      });
+    });
+  });
+
+  describe('--ruleset', () => {
+    describe('extends feature', () => {
+      it('should extend a valid relative ruleset', () => {
```

Comment:
hm.. I don't know. Some `it`s are in the form of `does` and others and in the form of `should do`. Maybe it would be good to unify these?